sdk/lib/opentelemetry/sdk/resources/resource.rb

```diff
@@ -57,7 +57,7 @@ def label_enumerator
         # @return [Resource] A new resource formed by merging the current resource
         #   with other
         def merge(other)
-          raise ArgumentError unless other.is_a?(Resource)
+          return self unless other.is_a?(Resource)
```

Comment:
I think this is totally reasonable, although I wonder, in Ruby 🦆 typing fashion, whether we might want to allow passing in a Hash or some other enumerable of pairs?

Meta question: what configuration should ultimately be supported for Resource? This seems to be an open question for the spec, not just for Ruby.